src/schema.jl

```diff
@@ -47,13 +47,93 @@ function validate(s::Schema)
     return true
 end
 
+function guess_type(value)
```

Comment:
Is there a plan to rebase on `Field.cast_value`?

Reply:
Assuming we are aiming for the same functionality as Pythons `_TypeGuesser` ([schema.py L280](https://github.com/frictionlessdata/tableschema-py/blob/master/tableschema/schema.py#L280)), I'm not sure why we would do this here.

If you are referring to the ability to refer to the `Field` from validation, that has now been added to this PR, e.g. [validate.jl L56](https://github.com/loleg/tableschema-jl/blob/94f5b12e09cf84d8afa82032c7288525d331a392/test/validate.jl#L56)